Legacy/bonej/src/main/java/org/bonej/plugins/ParticleCounter.java

```diff
@@ -2204,23 +2136,25 @@ private void joinStructures(final ImagePlus imp, final int[][] particleLabels,
 		}
 	}
 
-	private List<int[]> getNeighbourhoodCoordinates(final int phase,
-		final short[] voxel, final int w, final int h, final int d)
+	private static List<int[]> get26NeighbourhoodCoordinates(final short[] voxel,
+															final int[] bounds)
 	{
 		final int x0 = voxel[0];
 		final int y0 = voxel[1];
 		final int z0 = voxel[2];
-		if (phase == BACK) {
-			return Stream.of(new int[] { x0 - 1, y0, z0 }, new int[] { x0 + 1, y0,
-				z0 }, new int[] { x0, y0 - 1, z0 }, new int[] { x0, y0 + 1, z0 },
-				new int[] { x0, y0, z0 - 1 }, new int[] { x0, y0, z0 + 1 }).filter(
-					a -> withinBounds(a[0], a[1], a[2], w, h, d)).collect(Collectors
-						.toList());
-		}
+		return get26NeighbourhoodCoordinates(x0, y0, z0, bounds[0], bounds[1], bounds[2]);
+	}
```

Comment:
why not `return get26NeighbourhoodCoordinates( voxels[0],voxels[1],voxels[2],bounds[0],...` ?

Reply:
Rather I think this should be `(x0, y0, z0, w, h, d)`